packages/eventual-send/src/index.d.ts

```diff
@@ -18,6 +18,20 @@ export type EOnly<T> = T extends (...args: infer P) => infer R
     >
   : ERef<T>;
 
+type FilteredKeys<T, U> = {
+  [P in keyof T]: T[P] extends U ? P : never;
+}[keyof T];
```

Comment:
neat!

Reply:
> Could you translate this into an english description? I'm having trouble figuring it out.

For the type `FilteredKeys<T, U>` return a union of property names/symbols/numbers P for which the record element `T[P]`'s type extends `U`.

Given `const x = { a: 123, b: 'hello', c: 42, 49: () => {}, 53: 67 }`,

`FilteredKeys<typeof x, number>` is the type `'a' | 'c' | 53`.
`FilteredKeys<typeof x, string>` is the type `'b'`.
`FilteredKeys<typeof x, 42 | 67>` is the type `'c' | 53`.
`FilteredKeys<typeof x, boolean>` is the type `never`.